backend/Squirrel.Core/Squirrel.Core.BLL/Services/DependencyAnalyzer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Squirrel.Core.BLL.Interfaces;
+using System.Text.RegularExpressions;
+
+namespace Squirrel.Core.BLL.Services;
+
+public class DependencyAnalyzer : IDependencyAnalyzer
+{
+    private List<string> FindReferences(string spContent)
+    {
+        List<string> references = new List<string>();
+
+        // Used regular expressions to find references
+        Regex spRegex = new Regex(
+           @"(?<=\execute|exec\s+)(?:(?:([#@_a-z][#@_$\.0-9a-z]*)(?!\]))|[\[]([#@_a-z][#@_$\.0-9a-z\s]*)[\]])",
+            RegexOptions.IgnoreCase);
+
+        Regex functionRegex = new Regex(
+            @"\b\w+\.\w+\s*(?=\()",
+            RegexOptions.IgnoreCase);
+
+        // Got matches from both regex patterns
+        MatchCollection spMatches = spRegex.Matches(spContent);
+        MatchCollection functionMatches = functionRegex.Matches(spContent);
+
+        // Combined the results into a single list
+        references.AddRange(spMatches.Concat(functionMatches).Select(m => m.Value));
+
+        return references;
+    }
+
+    public List<string> AnalyzeDependencies(string spContent, List<string>? objectList = null)
+    {
+        List<string> references = FindReferences(spContent);
+
+        if (objectList == null)
```

Comment:
don't use == for null checks, because this operator could be overloaded. It is better to use pattern matching:
`objectList is null`